### PR TITLE
[ticket/16004] Add check-in for Emojis in Username

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16004

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16004

### CODE REVIEW

File: phpBB/includes/functions_user.php [hunk 1]
```diff
@@ -1709,6 +1709,14 @@ function validate_username($username, $allowed_username = false)
 		return false;
 	}
 
+	// The very first check is for
+	// out-of-bounds characters that are currently
+	// not supported by utf8_bin in MySQL
+	if (preg_match('/[\x{10000}-\x{10FFFF}]/u', $username))
+	{
+		return 'INVALID_EMOJIS_USERNAME';
```
Comment (VSEphpbb) [inline phpBB/includes/functions_user.php]: It should just return `INVALID_EMOJIS` sine `_USERNAME` is appended to the result of this function

**Comment by VSEphpbb** [comment]
Should test this in the validate_username_test test

File: phpBB/includes/functions_user.php (hunk 1 above)
Comment (3D-I) [reply phpBB/includes/functions_user.php]: Done, thx.

**Comment by 3D-I** [comment]
I will see to add such test therein.

**Comment by VSEphpbb** [comment]
Also, just a thought... Is it true Emoji are not allowed in a name? Why not?

**Comment by DavidIQ** [comment]
Well I think I'd rather support them than deny them. But perhaps that's something that we need to push off for a future version as it would require the username column data type to be changed, a likely expensive and time-consuming operation.

**Comment by VSEphpbb** [comment]
But it would be kind of hard to know what to call a user named: 🐸

**Comment by DavidIQ** [comment]
"Kermit"? "None of my business meme"? "Pepe"? "Trigger"? Take your pick 😆

**Comment by 3D-I** [comment]
If you allow emojis in usernames then many other tables will break, like Logs, Notifications etc.

**Comment by 3D-I** [comment]
The test is a courtesy of @mrgoldy :+1:

**Pull request merged**